src/type_/environment.rs

```diff
@@ -512,28 +511,24 @@ impl<'a, 'b> Environment<'a, 'b> {
 
     /// Increments an entity's usage in the current or nearest enclosing scope
     pub fn increment_usage(&mut self, name: &str) {
-        let mut cascade = None;
-        for scope in self.entity_usages.iter_mut().rev() {
-            match scope.get_mut(name) {
-                Some((EntityKind::PrivateTypeConstructor(type_name), _, usages)) => {
-                    // If a type constructor is used, we consider
-                    // its type also used
-                    if type_name != name {
-                        cascade = Some(type_name.clone());
-                    }
-                    *usages = true;
-                    break;
-                }
-                Some(val) => {
-                    val.2 = true;
-                    break;
+        let mut next_name = name.to_string();
+
+        while let Some((kind, _, used)) = self
+            .entity_usages
+            .iter_mut()
+            .rev()
+            .find_map(|entity| entity.get_mut(&next_name))
```

Comment:
This isn't an entity, it is a scope.
```suggestion
            .find_map(|scope| scope.get_mut(&next_name))
```

Reply:
✅